data/scripts/autogenerate_topld_variants.py

```diff
@@ -0,0 +1,143 @@
+# Autogenerate variants from TopLD that are not in FAVOR
+
+# Instructions:
+# 1. Install AWS CLI and configure it with appropriate credentials
+# 2. Run: pypy3 autogenerate_topld_variants.py
+# 3. The script will create one JSONL file for each chromosome with the missing variants
+# 4. Combine all files into a single file: cat chr*.jsonl > autogenerated_topld_variants.jsonl
+# 5. Remove duplicated lines: awk '!seen[$0]++' autogenerated_topld_variants.jsonl > autogenerated_topld_variants_uniq.jsonl
+# 6. Add keys and spdi + hgvs by using the `add_key_spdi_hgvs` function. It's a lot faster to generate the file above and then create the keys later.
+
+import os
+import pickle
+import json
+
+S3 = 's3://igvf-catalog-parsed-collections/variants/'
+FAVOR = 'favor_chrC_spdi_key_format.jsonl'
+TOPLDs = ['topld_eur_chrC_key.jsonl', 'topld_eas_chrC_key.jsonl',
+          'topld_afr_chrC_key.jsonl', 'topld_sas_chrC_key.jsonl']
+
+
+def add_key_spdi_hgvs(input_file, output_file):
+    from ga4gh.vrs.extras.translator import AlleleTranslator
+    from ga4gh.vrs.dataproxy import create_dataproxy
+    from biocommons.seqrepo import SeqRepo
+    import sys
+    sys.path.append(os.path.abspath(
+        os.path.join(os.path.dirname(__file__), '..')))
+    from adapters.helpers import build_spdi, build_hgvs_from_spdi
+
+    dp = create_dataproxy(
+        'seqrepo+file:///usr/local/share/seqrepo/2024-12-20')
+    seq_repo = SeqRepo('/usr/local/share/seqrepo/2024-12-20')
+    translator = AlleleTranslator(data_proxy=dp)
+
+    output = open(output_file, 'w+')
+
+    for line in open(input_file, 'r'):
+        data = json.loads(line)
+
+        chrm = data['chr']
+        pos = data['pos']
+        ref = data['ref']
+        alt = data['alt']
+
+        variation_type = 'SNP'
+        if len(ref) < len(alt):
+            variation_type = 'insertion'
+        elif len(ref) > len(alt):
+            variation_type = 'deletion'
+
+        spdi = build_spdi(
+            chrm,
+            pos,
+            ref,
+            alt,
+            translator,
+            seq_repo
+        )
+
+        hgvs = build_hgvs_from_spdi(spdi)
+        allele = translator.translate_from(spdi, 'spdi')
+        allele_vrs_digest = allele.digest
+
+        data['variantion_type'] = variation_type
+        data['spdi'] = spdi
+        data['hgvs'] = hgvs
+        data['_key'] = spdi if len(spdi) < 254 else allele_vrs_digest
```

Comment:
should add vrs_digest, organism or some other fields if possible that are common in other variants data.